ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/peers/DefaultPeerId.java

```diff
@@ -39,6 +41,11 @@ public Bytes32 keccak256() {
     return keccak256;
   }
 
+  @VisibleForTesting
+  public void setKeccak256(final Bytes32 keccak256) {
```

Comment:
Can we mock this where we need it for testing?  Instead of adding this setter to the production code?

Reply:
It has been mocked before, but the new paradigm of peer refresh requires that we have actual peers being created in the `createPeersInLastBucket` method of `PeerDiscoveryControllerTest`, so that we can query them for associated metadata.